src/Discord/WebSockets/Events/InteractionCreate.php

```diff
@@ -22,7 +24,45 @@ class InteractionCreate extends Event
      */
     public function handle(Deferred &$deferred, $data): void
     {
-        // do nothing with interactions - pass on to DiscordPHP-Slash
-        $deferred->resolve($this->factory->create(Interaction::class, $data, true));
+        $interaction = $this->factory->create(Interaction::class, $data, true);
+
+        if ($interaction->type == InteractionType::APPLICATION_COMMAND) {
+            $checkCommand = function ($command) use ($interaction, &$checkCommand) {
+                // Tries to cache the command
+                // possibly the laziest thing ive ever done - stdClass -> array
+                $cmd = json_decode(json_encode($command), true);
```

Comment:
If I remember correctly this was only done because the slash client didn't use the same `Part` as this library, and encoded with arrays instead of stdclasses. Might be able to get around this now?

Reply:
This is not necessary now, I just realized that caching command from Interaction Data wouldn't be good for commands with sub commands (options), due to interaction data only sending selected options and not all of available options in the command. Commands can be still manually fetched anyway, I'm Removing this.